src/proposals/Proposals.jl

```diff
@@ -245,5 +246,154 @@ function (prop::RStagger)(rng::AbstractRNG,
     return u, v, logl, ncall
 end
       
+"""
+    Proposals.Slice(;slices=5, scale=1)
+
+Propose a new live point by a series of random slices away from an existing live point.
+This is a standard _Gibbs-like_ implementation where a single multivariate slice is a combination of `slices` univariate slices through each axis.
+
+## Parameters
+- `slices` is the minimum number of slices
+- `scale` is the proposal distribution scale, which will update _between_ proposals.
+"""
+@with_kw mutable struct Slice <: AbstractProposal
+    slices = 5
+    scale = 1
+    # fmove = 0.9 this parameter to be included in Proposals.HSlice
+    # max_move = 100 ?? not sure if this parameter is to be included only in Proposals.HSlice or in all 3 Slice Proposals
+    
+    @assert slices ≥ 1 "Number of slices must be greater than or equal to 1"
+    @assert scale ≥ 0 "Proposal scale must be non-negative"
+end
 
+function (prop::Slice)(rng::AbstractRNG,
+                       point::AbstractVector,
+                       logl_star,
+                       bounds::AbstractBoundingSpace,
+                       loglike,
+                       prior_transform;
+                       kwargs...)
+    # setup
+    n = length(point)
+    slices_init = prop.slices
+    nc = nexpand = ncontract = 0
+    fscale = [] 
+    axlens = []
+    local idxs, r, u, u_prop, v_prop, logl_prop, logl_l, logl_r
+    
+    # modifying axes and computing lengths
+    axes = prop.scale * transpose(axes)  # scale based on past tuning
+    for axis in axes
+        append!(axlens, axis)
+    end
+    
```

Comment:
not yet

Reply:
@SaranjeetKaur in general you should probably not submit code here that you haven't actually run -- everything we see here should be after you try to run everything on your own computer and make sure it is achieving the goal of the PR. It's fine if there's bugs and stuff, but at the very least undefined variables like this could be caught very easily by just running all the code you write before committing it.

Reply:
After consulting the dynesty code as well as the polychord paper, `axes` here needs to be the Cholesky factorization of the covariance matrix of the bounding volume. For the Unit bounds, this is just `I`, for the ellipsoid (and by extension the multi-ellipsoid), I store the precision matrix `A`, so to get the correct basis space I'll have to do some linear algebra.

What I'll do is push to master a change that will let you call `taxes(bound)` which you can then do 
```julia
axes = prop.scale .* taxes(bound)'
axlens = [norm(@view(axes[i, :])) for i in Base.axes(axes, 1)]
```

Reply:
done